execution_engine/src/shared/transform.rs

```diff
@@ -54,8 +54,8 @@ impl From<CLValueError> for Error {
 
 /// Representation of a single transformation ocurring during execution.
 ///
-/// Note that all variants of [`Transform`] are commutative which means that a given collection of
-/// them can be executed in any order to produce the same end result.
+/// Note that all arithmetic variants of [`Transform`] are commutative which means that a given
```

Comment:
This still isn't precise IMO - why not simply say that all `Add*` variants are commutative? What does "arithmetic" stands for? is `AddNamedKeys` arithmetic? Also, if there is no other "arithmetic" operation that `Add` (like there's no `Subtract`) we shouldn't try to be too smart about it.